### ISSUE TITLE
Dispatch commandPath to a docker-container

### ISSUE THREAD

**Issue by waiting-for-dev** [neoclide/coc-solargraph#57]
First of all, thanks for maintaining this integration of solargraph with vim :heart: 

I'm trying to dispatch solargraph command to a docker container, but it seems not to be possible:

```yaml
"solargraph.commandPath": "docker-compose exec app bundle exec solargraph",
"solargraph.useBundler": false
```

I do have solargraph in the `Gemfile` and the command `docker-compose exec app bundle exec solargraph` works without problems in the console.

Thanks for any help.

**Comment by chemzqm** [comment]
I don't think that could work since solargraph in your docker can't read your files.

**Comment by waiting-for-dev** [comment]
Thanks for your answer.

It should be possible, as the files are accessible through a shared volume. They are accessible from the command line.

It seems more a problem about how the executable is resolved, as coc.nvim doesn't seem to recognize it and it prompts me to install the solargraph gem.

**Comment by demersus** [comment]
I had this working by using stdio as the solargraph.transport.  It is totally possible with your files mapped into the container.

I did create a script file named solargraph in my project bin folder that executes solargraph through docker-compose:

```
#!/bin/bash
compose_file=`dirname ${BASH_SOURCE[0]}`/../docker-compose.yml
docker compose -f $compose_file run --rm rails-app bundle exec solargraph "$@"
```

Make sure you set `solargraph.useBundler` to false
